Sources/TensorFlow/Layers/Convolutional.swift

```diff
@@ -485,3 +485,119 @@ public extension TransposedConv2D {
+    /// The element-wise activation function.
+    @noDerivative public let activation: Activation
+    /// The strides of the sliding window for spatial dimensions.
+    @noDerivative public let strides: (Int, Int)
+    /// The padding algorithm for convolution.
+    @noDerivative public let padding: Padding
+
+    /// Creates a `DepthwiseConv2D` layer with the specified filter, bias, activation function, strides, and
+    /// padding.
+    ///
+    /// - Parameters:
+    ///   - kernel: The 4-D convolution kernel.
+    ///   - bias: The bias vector.
+    ///   - activation: The element-wise activation function.
+    ///   - strides: The strides of the sliding window for spatial dimensions.
+    ///   - padding: The padding algorithm for convolution.
+    public init(
+        kernel: Tensor<Scalar>,
+        bias: Tensor<Scalar>,
+        activation: @escaping Activation,
+        strides: (Int, Int),
+        padding: Padding
+    ) {
+        self.kernel = kernel
+        self.bias = bias
+        self.activation = activation
+        self.strides = strides
+        self.padding = padding
+    }
+
+    /// Returns the output obtained from applying the layer to the given input.
+    ///
+    /// - Parameter input: The input to the layer.
+    /// - Returns: The output.
+    @differentiable
+    public func callAsFunction(_ input: Tensor<Scalar>) -> Tensor<Scalar> {
+        return activation(depthwiseConv2D(input, kernel: kernel,
+                          strides: (1, strides.0, strides.1, 1),
+                          padding: padding) + bias)
+    }
+}
+
+public extension DepthwiseConv2D {
+    /// Creates a `depthwiseConv2D` layer with the specified kernel shape, strides, padding, and
+    /// element-wise activation function. The kernel tensor is initialized using Glorot uniform
+    /// initialization with the specified generator. The bias vector is initialized with zeros.
+    ///
+    /// - Parameters:
+    ///   - kernelShape: The shape of the 4-D convolution kernel.
+    ///   - strides: The strides of the sliding window for spatial/spatio-temporal dimensions.
+    ///   - padding: The padding algorithm for convolution.
+    ///   - activation: The element-wise activation function.
+    ///   - generator: The random number generator for initialization.
+    ///
+    /// - Note: Use `init(filterShape:strides:padding:activation:seed:)` for faster random
+    ///   initialization.
+    init<G: RandomNumberGenerator>(
+        kernelShape: (Int, Int, Int, Int),
+        strides: (Int, Int) = (1, 1),
+        padding: Padding = .valid,
+        activation: @escaping Activation = identity,
+        generator: inout G
+    ) {
+        let kernelTensorShape = TensorShape([
+            kernelShape.0, kernelShape.1, kernelShape.2, kernelShape.3])
+        self.init(
+            kernel: Tensor(glorotUniform: kernelTensorShape, generator: &generator),
+            bias: Tensor(zeros: TensorShape([kernelShape.3])),
```

Comment:
Instead of calling the `TensorShape` initializer directly, always use array literal initialization when there is a contextual type. `Tensor`'s `init(zeros:)` initializer takes a `TensorShape`, therefore the contextual type is `TensorShape` already.
```suggestion
            bias: Tensor(zeros: [kernelShape.3]),
```

Reply:
@rxwei most layers extensions` bias initialization are wrapped in a `TensorShape()`

Reply:
They should be changed.

Reply:
done in #249